frontend/src/routes/Credentials/AddCredentials/AddCredentials.tsx

```diff
@@ -237,16 +238,26 @@ export function AddCredentialPageContent(props: { providerConnection: ProviderCo
     const history = useHistory()
     const [featureGates] = useRecoilState(featureGatesState)
     const discoveryFeatureGate = featureGates.find((fg) => fg.metadata.name === 'open-cluster-management-discovery')
-
+    const [mchNamespace, setmchNamespace] = useState<string>()    
     const isEditing = () => props.providerConnection.metadata.name !== ''
     const alertContext = useContext(AcmAlertContext)
-
     const [providerConnection, setProviderConnection] = useState<ProviderConnection>(
         JSON.parse(JSON.stringify(props.providerConnection))
     )
     // useEffect(() => {
     //     setProviderConnection(JSON.parse(JSON.stringify(props.providerConnection)))
     // }, [props.providerConnection])
+    useEffect(() => {
+        const result = listMultiClusterHubs()
```

Comment:
So I think it's better to just add MCH to our watch list; you'll have to add the list permission to the chart here:

https://github.com/open-cluster-management/console-chart/blob/main/stable/console-chart/templates/console-clusterrole.yaml#L33

---

After that we can store the MCH in the Recoil cache and just read from it whenever we need it; that will help us not having to fetch the MCH over and over and simplify the tests a lot so you don't have to deal with the nocks.